pkg/component/vpa/vpa.go

```diff
@@ -244,7 +244,10 @@ func (v *vpa) emptyMutatingWebhookConfiguration() *admissionregistrationv1.Mutat
 		suffix = "target"
 	}
 
-	return &admissionregistrationv1.MutatingWebhookConfiguration{ObjectMeta: metav1.ObjectMeta{Name: "vpa-webhook-config-" + suffix}}
+	// The order in which MutatingWebhooks are called is determined alphabetically. This name intentionally starts with
+	//'zzz', such that the webhook is called after all other webhooks which inject containers. All containers injected
+	// by webhooks that are called _after_ the vpa webhook will not be under control of vpa.
```

Comment:
> Is this generally something we want to duplicate? If you want to understand why a name is intentionally set in a certain way, you'd look up the code that does it, right?

Not in general, but in this case, the name is quite special/noticeable. I think it would be helpful for people (including end-users since this affects them as well) to find this information in an annotation instead of trying to look it up in the code base (which is non-trivial if you don't know where to start).

> If I'm on the other hand thinking about the next time we change this name or arrangement, the documentation replicated in the annotation is prone to be left over and forgotten.

This is rather a theoretical case - it is quite unlikely that we have to change the name again in the foreseeable future. Nevertheless, if you don't want to duplicate it, the Go comment can also refer to the annotation if developers are stumbling across this place.

Reply:
done.